tlspuffin/harness/openssl/src/put.c

```diff
@@ -383,18 +357,75 @@ static AGENT make_agent(SSL_CTX *ssl_ctx, const TLS_AGENT_DESCRIPTOR *descriptor
     return agent;
 }
 
-static RESULT_CODE result_code(AGENT agent, int retcode)
+static RESULT get_result(AGENT agent, int retcode, bool allow_would_block)
 {
     int ssl_ecode = SSL_get_error(agent->ssl, retcode);
+
+    char *reason = get_error_reason();
+    char *error_type;
+    RESULT_CODE res = RESULT_OK;
+
     switch (ssl_ecode)
     {
     case SSL_ERROR_NONE:
-        return RESULT_OK;
-
+        error_type = strdup("no error");
+        break;
+    case SSL_ERROR_ZERO_RETURN:
+        error_type = strdup("SSL_ERROR_ZERO_RETURN");
+        break;
+    case SSL_ERROR_WANT_CONNECT:
```

Comment:
looks like this is rather a `IO_WOULD_BLOCK` according to https://docs.openssl.org/1.1.1/man3/SSL_get_error/#return-values
Might also be the case for other types of errors? (I did not check all of them)

Reply:
Done